src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -1788,6 +1788,22 @@ private void InitializePreferenceLocations()
             {
                 PreferenceSettings.TemplateFilePath = pathManager.DefaultTemplatesDirectory;
             }
+            var supportedLocales = Configurations.SupportedLocaleDic.Values.ToList<string>();
+
+            //Get the last part of the template path e.f. if the path is C:\ProgramData\Dynamo\Dynamo Core\templates\en-US then currentPathLocale = en-US
+            var pathParts = PreferenceSettings.TemplateFilePath.Split("\\");
+            var currentPathLocale = pathParts[pathParts.Count() - 1];
```

Comment:
There should be a Last() pointer, no?

Reply:
Fixed in the next commit: https://github.com/DynamoDS/Dynamo/pull/15657/commits/73fb8229b0af4e2d671467b6d95758dcb6605a67